scripts/sbom_scraper.sh

```diff
@@ -402,67 +477,73 @@ END
 # xmllint complains about a double import of the spdx schema, but we have to import via
 # the wrapper to set the schema location to a local file, as xmllint fails to download
 # them from the internet as they are https
-xmllint "$PATCHED_OUTPUT" --schema "$SCRIPTDIR"/cyclonedx-wrapper.xsd --noout 2>&1 | grep -Fv "Skipping import of schema located at 'http://cyclonedx.org/schema/spdx' for the namespace 'http://cyclonedx.org/schema/spdx'"
-[ "${PIPESTATUS[0]}" -ne 0 ] && exit "${PIPESTATUS[0]}"
+_=$(xmllint "$PATCHED_OUTPUT" --schema "$SCRIPTDIR"/cyclonedx-wrapper.xsd --noout 2>&1 | grep -Fv "Skipping import of schema located at 'http://cyclonedx.org/schema/spdx' for the namespace 'http://cyclonedx.org/schema/spdx'")
+[ "${PIPESTATUS[0]}" -ne 0 ] && cat "${PATCHED_OUTPUT}" && exit "${PIPESTATUS[0]}"
 
-# ----------------------------------------------------------------------------
-# Handle client id and secrets for SBOM scraper via App registrations
-# ----------------------------------------------------------------------------
-HTTP_STATUS=""
-# get token
-log "Get token ..."
-HTTP_STATUS=$(curl -sS -w "%{http_code}" \
-    -o "${TEMPDIR}/access_token" \
-    --data-urlencode "grant_type=client_credentials" \
-    --data-urlencode "client_id=${CLIENT_ID}" \
-    --data-urlencode "client_secret=${SECRET}" \
-    "${URL}/archivist/iam/v1/appidp/token")
-if [ "${HTTP_STATUS}" != "200" ]
+if [ "${UPLOAD}" = "false" ]
 then
-    log "Get token failure ${HTTP_STATUS}"
-    exit 2
-fi
+    # not uploading - just output the xml
+    cat "${PATCHED_OUTPUT}"
+else
+    # ----------------------------------------------------------------------------
+    # Handle client id and secrets for SBOM scraper via App registrations
+    # ----------------------------------------------------------------------------
+    HTTP_STATUS=""
+    # get token
+    log "Get token ..."
+    HTTP_STATUS=$(curl -sS -w "%{http_code}" \
+        -o "${TEMPDIR}/access_token" \
+        --data-urlencode "grant_type=client_credentials" \
+        --data-urlencode "client_id=${CLIENT_ID}" \
+        --data-urlencode "client_secret=${SECRET}" \
+        "${URL}/archivist/iam/v1/appidp/token")
+    if [ "${HTTP_STATUS}" != "200" ]
+    then
+        log "Get token failure ${HTTP_STATUS}"
+        exit 2
+    fi
 
-TOKEN=$(jq -r .access_token "${TEMPDIR}"/access_token )
+    TOKEN=$(jq -r .access_token "${TEMPDIR}"/access_token )
```

Comment:
is there some mixed tab/spaces here ?

Reply:
ah, no it's all space - I just indented the whole block as it only happens if we run without -n